pkg/contracts/build.go

```diff
@@ -0,0 +1,163 @@
+package contracts
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/CoreumFoundation/coreum-tools/pkg/libexec"
+	"github.com/CoreumFoundation/coreum-tools/pkg/logger"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// BuildConfig provides params for the building stage.
+type BuildConfig struct {
+	// NeedOptimizedBuild will enable running a Docker container that builds a very optimized and predictable build.
+	// This option must be enabled for any deployment-oriented build.
+	NeedOptimizedBuild bool
+}
+
+// Build implements logic for "contracts build" CLI command
+func Build(ctx context.Context, workspaceDir string, config BuildConfig) error {
+	log := logger.Get(ctx)
+
+	if err := ensureBuildTools(ctx, config.NeedOptimizedBuild); err != nil {
+		err = errors.Wrap(err, "not all build dependencies are installed")
+		return err
```

Comment:
Is it a code style you are using across all codebase? Like instead of just returning wrapped error